packages/playground/client/src/index.ts

```diff
@@ -95,28 +104,46 @@ export async function startPlaygroundWeb({
 		progressbar: !disableProgressBar,
 	});
 	progressTracker.setCaption('Preparing WordPress');
-	// Set a default blueprint if none is provided.
+
+  // Set a default blueprint if none is provided.
 	if (!blueprint) {
 		blueprint = {
 			phpExtensionBundles: ['kitchen-sink'],
 		};
 	}
+
 	const compiled = compileBlueprint(blueprint, {
 		progress: progressTracker.stage(0.5),
 		onStepCompleted: onBlueprintStepCompleted,
 	});
-	const playground = await doStartPlaygroundWeb(
-		iframe,
-		setQueryParams(remoteUrl, {
-			php: compiled.versions.php,
-			wp: compiled.versions.wp,
-			['sapi-name']: sapiName,
-			['php-extension']: compiled.phpExtensions,
-			['networking']: compiled.features.networking ? 'yes' : 'no',
-			'site-slug': siteSlug,
-		}),
-		progressTracker
-	);
+
+	await new Promise((resolve) => {
+		iframe.src = remoteUrl;
+		iframe.addEventListener('load', resolve, false);
+	});
+
+	// Connect the Comlink API client to the remote worker,
+	// boot the playground, and run the blueprint steps.
+	const playground = consumeAPI<PlaygroundClient>(
+		iframe.contentWindow!,
+		iframe.ownerDocument!.defaultView!
+	) as PlaygroundClient;
+	await playground.isConnected();
+	progressTracker.pipe(playground);
+	const downloadPHPandWP = progressTracker.stage();
+	await playground.onDownloadProgress(downloadPHPandWP.loadingListener);
+	await playground.boot({
+		mounts,
+		sapiName,
+		shouldInstallWordPress,
+		phpVersion: compiled.versions.php,
+		wpVersion: compiled.versions.wp,
+		phpExtensions: compiled.phpExtensions,
+		withNetworking: compiled.features.networking,
+	});
```

Comment:
This is unfortunately triggering a DataCloneError in Safari. The cause appears to be the `mounts` property which is an array of [FileSystemDirectoryHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) instances.

Reply:
Correction:
The cause appears to be the `mounts` property which is an array of MountDescriptors that contain [FileSystemDirectoryHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) references.

Reply:
It's possible I'm wrong about the cause of the DataCloneError in Safari. There's a bit in the MDN docs that says:

> Note: Objects based on [FileSystemHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle) can also be serialized into an [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) database instance, or transferred via [postMessage()](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).

https://developer.mozilla.org/en-US/docs/Web/API/File_System_API#sect4

Reply:
Confirmed that `postMessage()` with FileSystemDirectoryHandle does produce a DataCloneError.

Based on the File System spec, the reason could be:
> If serialized.[[Origin]] is not [same origin](https://html.spec.whatwg.org/multipage/browsers.html#same-origin) with value’s [relevant settings object](https://html.spec.whatwg.org/multipage/webappapis.html#relevant-settings-object)'s [origin](https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-origin), then [throw](https://webidl.spec.whatwg.org/#dfn-throw) a "[DataCloneError](https://webidl.spec.whatwg.org/#datacloneerror)" [DOMException](https://webidl.spec.whatwg.org/#idl-DOMException).

https://fs.spec.whatwg.org/#ref-for-deserialization-steps

Reply:
Strange. From the Safari dev tools console:
```
> h = await window.navigator.storage.getDirectory();
< undefined
[Info] FileSystemDirectoryHandle {getFileHandle: function, getDirectoryHandle: function, removeEntry: function, resolve: function, entries: function, …}
> window.postMessage(h)
< DataCloneError: The object can not be cloned.
```

AFAICT, there shouldn't be an inconsistent origin there.

Reply:
There is an existing bug report:
> structuredClone throws for FileSystemHandles (Origin Private File System)

https://bugs.webkit.org/show_bug.cgi?id=256712

Reply:
That is a great catch and debugging, hats off to you 🙇  In this case, I'd downgrade the entire interface to support Safari until they fix that bug, and do something like `const handleId = worker.acquireOPFSDirectoryHandle()` and never move the actual object across workers.

Reply:
@adamziel, I started looking at this and wasn't quite sure what you imagined for
```
const handleId = worker.acquireOPFSDirectoryHandle()
```

But I think we should be able to just pass the mountpoint back and forth and just have a function to take a mountpoint and return a directory handle. Please let me know if you are envisioning something else.

Reply:
@adamziel correction: We'd pass OPFS path instead of `FileSystemDirectoryHandle` and use the OPFS path to resolve the `FileSystemDirectoryHandle` when needed.

Reply:
This should be resolved now. The MountDescriptor interface no longer has a FileSystemDirectoryHandle and looks like:
```ts
export interface MountDescriptor {
	mountpoint: string;
	opfsPath: string;
	initialSyncDirection: 'opfs-to-memfs' | 'memfs-to-opfs';
}
```